sbt-bridge/test/xsbt/CompileProgressReportTest.scala

```diff
@@ -0,0 +1,51 @@
+package xsbt
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class CompileProgressReportTest:
+
+  @Test
+  def testStartUnitReports(): Unit =
+    val source1 =
+      """class A {
+        |
+        |}
+        |""".stripMargin
+
+    val compilerForTesting = new ScalaCompilerForUnitTesting
+    val progress = compilerForTesting.compileSrcs(source1)._3
+
+    val expectedProgress = Seq(
+      ("Test-0-0.scala", "typer (parsing)"),
+      ("Test-0-0.scala", "typer (typechecking)"),
+      ("Test-0-0.scala", "typer (checking java)"),
+      ("Test-0-0.scala", "inlinedPositions"),
+      ("Test-0-0.scala", "sbt-deps"),
+      ("Test-0-0.scala", "posttyper"),
+      ("Test-0-0.scala", "sbt-api"),
+      ("Test-0-0.scala", "pickler"),
+      ("Test-0-0.scala", "inlining"),
+      ("Test-0-0.scala", "postInlining"),
+      ("Test-0-0.scala", "staging"),
+      ("Test-0-0.scala", "pickleQuotes"),
+      ("Test-0-0.scala", "MegaPhase{firstTransform, checkReentrant, elimPackagePrefixes, cookComments, checkStatic, betaReduce, inlineVals, expandSAMs}"),
```

Comment:
These "MegaPhase" phase names are not ideal, but I don't know how to better report them ATM, taking into account that all subphases are reported for each file

Reply:
Maybe just the first name, e.g. "miniphases following firstTransform"?

Reply:
Hmm, not sure if there should be information loss.
This could probably be done on the client-side, by parsing.
It has quite a simple structure for that.

Reply:
Also, maybe we could create meaningful names for each MegaPhase, and report it?
I just do not have enough expertise in the compiler to give proper names to them.

Reply:
> Also, maybe we could create meaningful names for each MegaPhase, and report it?

If we could, we would just put all mini-phases in a single MegaPhase (since that would result in a single pass over the AST), but because of implementation details that's not possible, so we split them into the minimum number of MegaPhases we can, but these MegaPhases are not meaningful on their own, so there's no useful name we can give them. In fact, I don't think the name of phases in general is useful information for end users, you could just as well use numbers ("Running phase 1/6 ...") or nothing at all ("Compiling ..."), since the number isn't very useful either.

Reply:
We do not print the information about the phases as text anywhere.
It's only reported in the Compilation Charts 
https://blog.jetbrains.com/scala/2021/04/07/intellij-scala-plugin-2021-1/#phases-and-units-in-compilation-charts

Yes, most of the users are not interested in this, but there were some users (Scala 2), dealing with macro and compiler internals, which reported their interest in this kind of information.

Reply:
Anyway, until I can call `advance` method, this method call will tell IDE that the compilation process is not stuck.
(currently, compilation can take minutes without any indication of progress)
Any value will be fine.